### ISSUE TITLE
Need to re-think CVE and CVSS

### ISSUE THREAD

**Issue by pagbabian-splunk** [ocsf/ocsf-schema#328]
Currently we have split the usage of `cve_uids` array of `string`s in the `malware` object while the `CVSS Scores` attribute of type `cvss` is part of the `Malware` profile.  These attributes should be more aligned as typically a CVE uid will have a CVSS score.  Further, a CVE uid may have different scores depending on the version of CVSS.

Our current `CVSS Scores` object tries to redefine the [https://www.first.org/cvss/](CVSS) Metrics and associated values as a series of OCSF enums (that do not follow the enum conventions).  As with MITRE ATT&CK we should refrain from redefining a dynamic standard such as CVE or CVSS but rather refer to them using invariant values.

For example, with CVE-2013-0375 (MySql Injection) there are multiple CVSS based scores, 2.0 and 3.1.  Each score is broken down into a set of Metrics and associated Values.

Rather than split our CVE and CVSS, we should pair their objects so that they can represent the vulnerabilities in the same was as they are published and discovered.

Following the pattern we used with ATT&CK, we should revisit our current CVSS Scores object, and adopt the Metrics and Values from the external standard.

Metrics vary per CVSS version.  E.g. for 2.0: Access Vector with Values of Local, Adjacent Network, Network.  For 3.1: Attack Vector with Values of Network, Adjacent, Local, Physical.

These are all captured well within our CVSS object but as a version independent set of custom enums.

Along with combining CVE, CVSS version, CVSS Metric and CVSS Value, we should consider using a version context name-based enum that follows the CVSS standard, rather than redefine as custom enums.

**Comment by floydtree** [comment]
Interesting observation.

Have you taken a look at the `vulnerability` [object](https://schema.ocsf.io/objects/vulnerability), used as a vulnerabilities array in the security_finding class. This object is what contextually relates CVE id and the corresponding CVSS scores. (Every vulnerability will have an identifier/s, i.e. CVE ids and each CVE Id will have the corresponding CVSS scores). Point being, the current combining factor is the vulnerability object.

We can probably rethink if the Malware object itself requires its own field to represent CVE ids, and try to utilize the vulnerability object instead.

As for the CVSS object, 
1. discarding the fact that it doesn't follow OCSF enum conventions, I think the object is designed in a way that it can be used to represent data for all the current CVSS versions. The CVSS version can be represented using the `version` field and corresponding metrics-fields can be populated accordingly. 
2. Are you suggesting string enums instead of the current int enums?

@irakledibm What do you think about the observations re. the CVSS object?

**Comment by irakledibm** [comment]
You are right Rajas, CVSS object that we have is representing https://www.first.org/cvss/ exactly. Only difference is that optional attributes are supporting all CVSS versions, based on specified version field appropriate attributes can be validated/pulled. Descriptions usually specify to which version particular attribute belongs. As of enum, I think it is not following conventions since this object was added before enum convention was finalized, we can update it to latest standard.

**Comment by irakledibm** [comment]
And I agree that a vulnerability object provides pairing of cve id and cvss score. Maybe we should introduce CVE details object since vulnerability may or may not be CVE. Then, in malware object we may include list of cve objects instead of ids.

**Comment by irakledibm** [comment]
Another thing, our vulnerability details object is missing few useful attributes, such as vuln type(s),  publish date, update date etc. Maybe we can discuss it, should we schedule findings group meeting?

**Comment by floydtree** [comment]
>And I agree that a vulnerability object provides pairing of cve id and cvss score. Maybe we should introduce CVE details object since vulnerability may or may not be CVE. Then, in malware object we may include list of cve objects instead of ids.

Agreed, this could potentially be the path forward. Further, the CVSS object can then be nested inside this new CVE object, that will cement the context properly. Both Malware and Vulnerability objects can then utilize this CVE object.

>Another thing, our vulnerability details object is missing few useful attributes, such as vuln type(s), publish date, update date etc. Maybe we can discuss it, should we schedule findings group meeting?

Agreed, a call seems necessary. I am out until mid November though, other members of the Findings group can chime in?
